numpy/core/tests/test_multiarray.py

```diff
@@ -2326,6 +2326,22 @@ def test_dot_type_mismatch(self):
         assert_raises(TypeError, np.dot, c, A)
         assert_raises(TypeError, np.dot, A, c)
 
+    def test_dot_out_mem_overlap(self):
+        np.random.seed(1)
+
+        for dtype in [np.object_, np.float32, np.complex128, np.int64]:
```

Comment:
There are two code paths --- blas and non-blas, so you need at least two dtypes.